app/components/project-geometries/modes/draw.js

```diff
@@ -137,19 +157,47 @@ export default class DrawComponent extends Component {
 
     draw.setFeatureProperty(firstFeature.id, 'label', label);
 
+    // update special polygon feature flag used to require users to label their polygons
+    this.set('selectedFeature.features.firstObject.missingLabel', false);
+
     // this triggers an update that renders the new label as mutated above to show up in the selected feature
     // see https://github.com/mapbox/mapbox-gl-draw/blob/master/docs/API.md#events
     this.drawStateCallback();
   }
 
   @action
   handleDrawButtonClick() {
+    const currentMode = this.map.draw.drawInstance.getMode();
+    const { features: [firstFeature] } = this.get('selectedFeature') || {features: []};
+
+    if(!currentFeatureIsComplete(currentMode, firstFeature)) {
+      // set special polygon feature flag, used to require used to label their polygons in draw mode
+      this.set('selectedFeature.features.firstObject.missingLabel', true);
+      // block mode switch
+      return;
+    }
+
     this.map.draw.drawInstance.changeMode('draw_polygon');
     this.set('tool', 'draw_polygon');
   }
 
   @action
   handleAnnotation(mode) {
+    // require a user to finish drawing their polygon
+    const currentMode = this.map.draw.drawInstance.getMode();
+    if (currentMode === 'draw_polygon') {
+      return;
+    }
+
+    // require a user to label their finished polygon
+    const { features: [firstFeature] } = this.get('selectedFeature');
+    if(!currentFeatureIsComplete(currentMode, firstFeature)) {
+      // set special polygon feature flag, used to require used to label their polygons in draw mode
+      this.set('selectedFeature.features.firstObject.missingLabel', true);
```

Comment:
We should get some unit testing in here to check that this does what we expect (mutates the first element of the features array)